modules/nf-core/krakenuniq/preloadedkrakenuniq/main.nf

```diff
@@ -0,0 +1,118 @@
+process KRAKENUNIQ_PRELOADEDKRAKENUNIQ {
+    tag "$meta.id"
+    label 'process_high'
+
+    conda (params.enable_conda ? "bioconda::krakenuniq=1.0.0" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/krakenuniq:1.0.0--pl5321h19e8d03_0':
+        'quay.io/biocontainers/krakenuniq:1.0.0--pl5321h19e8d03_0' }"
+
+    input:
+    tuple val(meta), path(fastqs)
+    path  db
+    val ram_chunk_size
+    val save_output_fastqs
+    val report_file
+    val save_output
+
+    output:
+    tuple val(meta), path('*.classified{.,_}*')     , optional:true, emit: classified_reads_fastq
+    tuple val(meta), path('*.unclassified{.,_}*')   , optional:true, emit: unclassified_reads_fastq
+    tuple val(meta), path('*classified.txt')        , optional:true, emit: classified_assignment
+    tuple val(meta), path('*report.txt')                           , emit: report
+
+    path "versions.yml"                                            , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def args2 = task.ext.args ?: ''
+
+    def paired       = meta.single_end ? "" : "--paired"
+    def classified   = meta.single_end ? '"\$PREFIX".classified.fastq'   : '"\$PREFIX".classified#.fastq'
+    def unclassified = meta.single_end ? '"\$PREFIX".unclassified.fastq' : '"\$PREFIX".unclassified#.fastq'
+    def classified_option = save_output_fastqs ? "--classified-out ${classified}" : ""
+    def unclassified_option = save_output_fastqs ? "--unclassified-out ${unclassified}" : ""
+    def output_option = save_output ? '--output "\$PREFIX".krakenuniq.classified.txt' : ""
+    def report = report_file ? '--report-file "\$PREFIX".krakenuniq.report.txt' : ""
+    def compress_reads_command = save_output_fastqs ? "gzip *.fastq" : ""
+
+    """
+    krakenuniq \\
+        $args \\
+        --db $db \\
+        --preload $ram_chunk_size \\
+        --threads $task.cpus;
+
+    for fastq in ${fastqs.join(' ')}; do \\
+        PREFIX=\$(echo \$fastq);
+        krakenuniq \\
+            --db $db \\
+            --threads $task.cpus \\
+            $report \\
+            $output_option \\
+            $unclassified_option \\
+            $classified_option \\
+            $output_option \\
+            $paired \\
+            $args2 \\
+            \$fastq;
+    done
+
```

Comment:
```suggestion

    gzip *.fastq

```

Reply:
`pigz` is not available in the container by any chance? It would allow multi-threaded compression. Please use the `--no-name` option (option might change depending on the gzip version) that prevents including the timestamp in the compressed file.

Reply:
Actually, it looks like the gzip from the container automatically excludes the timestamp, however, I'm not sure the same is true for whatever is provided by conda/platform bash.

Reply:
Since `pigz` is not an option unless we'd use a mulled container, we can at least parallelize gzip using the technique presented in this answer https://unix.stackexchange.com/a/216475 under the heading "Parallel runs in N-process batches" using the `task.cpus` for N.

Reply:
I would rather stick to just the one by one method for now, as 1) it's only if reads are requsted, and 2) inserting complicated(ish) bash might be a bit risky without proper testing (and I _really_ want to get first release of taxprofiler done 😬)

Reply:
Agreed. I only suggested it since all sequence files will be compressed sequentially. which will be rather slow.